lib/train/plugins/base_connection.rb

```diff
@@ -37,6 +37,25 @@ def initialize(options = nil)
       end
     end
 
+    # Returns cached client if caching enabled. Otherwise returns whatever is
+    # given in the block.
+    #
+    # @example
+    #
+    #   def demo_client
+    #     cached_client(:api_call, :demo_connection) do
+    #       DemoClient.new(args)
+    #     end
+    #   end
+    #
+    # @param [symbol] type one of [:api_call, :file, :command]
+    # @param [symbol] key for your cached connection
+    def cached_client(type, key)
+      return yield unless cache_enabled?(type)
+
+      @cache[type][key] ||= yield
```

Comment:
`@cache.dig(type, key) ||= yield`?

Reply:
I wasn't sure if you could do assignment when using dig.

Reply:
That doesn't work. You aren't allowed to do assignment from the return of `dig`. 

```
train/lib/train/plugins/base_connection.rb:56: syntax error, unexpected tOP_ASGN,
expecting keyword_end (SyntaxError)
      @cache.dig(type, key) ||= yield
```